app/services/hyrax/resource_characterizer.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Hyrax
+  ##
+  # Characterizes a Valkyrie based FileSet
+  class ResourceCharacterizer
+    ##
+    # @!attribute [rw] source
+    #   @return [#characterize]
+    attr_accessor :source
+
+    ##
+    # @param source the object to characterize
+    def initialize(source:)
+      self.source = source
+    end
+
+    ##
+    # @return [void]
+    #
+    # @raise [RuntimeError] if visibility propogation fails
+    def characterize
+      Hydra::Works::CharacterizationService.run(characterization_proxy, filepath)
+      Rails.logger.debug "Ran characterization on #{characterization_proxy.id} (#{characterization_proxy.mime_type})"
+      characterization_proxy.alpha_channels = channels(filepath) if source.image? && Hyrax.config.iiif_image_server?
+      persist
```

Comment:
Similar to the above comment, I wonder about the utility of moving the persistence code into this method.